File: split/newcharacter.pas
function newcharacter(f: internalfontnumber; c: eightbits): halfword;
label
  10;
var
  p: halfword;
begin
  if fontbc[f] <= c then
    if fontec[f] >= c then
      if (fontinfo[charbase[f] + c + 0].qqqq.b0 > 0) then
      begin
        p := getavail;
        mem[p].hh.b0 := f;
        mem[p].hh.b1 := c + 0;
        newcharacter := p;
        goto 10;
      end;
  charwarning(f, c);
  newcharacter := 0;
  10: ;
end;

File: cpp/newcharacter.cpp
#include "newcharacter.h"
#include "getavail.h"
#include "charwarning.h"

halfword newcharacter(internalfontnumber f, eightbits c)
{
	if (fontbc[f] <= c)
		if (fontec[f] >= c)
			if ((skip_byte(char_info(f, c)) > 0))
			{
				auto p = getavail();
				type(p) = f;
				subtype(p) = c;
				return p;
			}
	charwarning(f, c);
	return 0;
}
